src/frequenz/sdk/timeseries/periodic_feature_extractor.py

```diff
@@ -0,0 +1,443 @@
+# License: MIT
+# Copyright © 2023 Frequenz Energy-as-a-Service GmbH
+
+"""
+A class that extracts features out of historical data.
+
+It creats a profile from periodically occurring
+windows in a buffer of historical data.
+
+The profile is created out of all windows that are fully
+contained in the underlying buffer with the same
+start and end time modulo a fixed period.
+
+**Example:**
+
+Consider a timeseries $T$ of historical data and
+subseries $S_i$ of $T$ all having the same size $l$.
+and the same fixed distance $p$ called period,
+where period of two subwindows is defined as the distance
+of two points at the same position within the subwindows.
+
+This class calculates a statistical profile S over all $S_i$,
+i.e. the value of S at position i is calculated by performing
+a certian calculation, e.g. an average, over all values of
+$S_i$ at position i.
```

Comment:
Some format improvements. And I replaced the `S` in `profile S over` and `value of S at position` to `$S$`, not sure if that's correct.

```suggestion
Example:

    Consider a timeseries $T$ of historical data and
    sub-series $S_i$ of $T$ all having the same size $l$.
    and the same fixed distance $p$ called period,
    where period of two sub-windows is defined as the distance
    of two points at the same position within the sub-windows.

    This class calculates a statistical profile $S$ over all $S_i$,
    i.e. the value of $S$ at position i is calculated by performing
    a certain calculation, e.g. an average, over all values of
    $S_i$ at position i.
```